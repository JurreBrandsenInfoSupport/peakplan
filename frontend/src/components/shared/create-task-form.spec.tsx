import { render } from "@testing-library/react";
import { describe, expect, it, vi } from "vitest";
import CreateTaskForm from "./create-task-form";

describe("CreateTaskForm", () => {
  it("renders correctly", () => {
    const handleCreateTask = vi.fn(() => Promise.resolve());
    const component = render(
      <CreateTaskForm onCreateTask={handleCreateTask} />,
    );

    expect(component).toMatchSnapshot();
  });
});
