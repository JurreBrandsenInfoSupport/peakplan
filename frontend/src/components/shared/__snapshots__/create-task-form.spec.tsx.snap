// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`CreateTaskForm > renders correctly 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <form
        class="space-y-4"
      >
        <div
          class="grid gap-2"
          data-slot="form-item"
        >
          <label
            class="flex items-center gap-2 text-sm leading-none font-medium select-none group-data-[disabled=true]:pointer-events-none group-data-[disabled=true]:opacity-50 peer-disabled:cursor-not-allowed peer-disabled:opacity-50 data-[error=true]:text-destructive"
            data-error="false"
            data-slot="form-label"
            for="«r0»-form-item"
          >
            Title
          </label>
          <input
            aria-describedby="«r0»-form-item-description"
            aria-invalid="false"
            class="file:text-foreground placeholder:text-muted-foreground selection:bg-primary selection:text-primary-foreground dark:bg-input/30 border-input flex h-9 w-full min-w-0 rounded-md border bg-transparent px-3 py-1 text-base shadow-xs transition-[color,box-shadow] outline-none file:inline-flex file:h-7 file:border-0 file:bg-transparent file:text-sm file:font-medium disabled:pointer-events-none disabled:cursor-not-allowed disabled:opacity-50 md:text-sm focus-visible:border-ring focus-visible:ring-ring/50 focus-visible:ring-[3px] aria-invalid:ring-destructive/20 dark:aria-invalid:ring-destructive/40 aria-invalid:border-destructive"
            data-slot="form-control"
            id="«r0»-form-item"
            name="title"
            placeholder="Task title"
            value=""
          />
        </div>
        <div
          class="grid gap-2"
          data-slot="form-item"
        >
          <label
            class="flex items-center gap-2 text-sm leading-none font-medium select-none group-data-[disabled=true]:pointer-events-none group-data-[disabled=true]:opacity-50 peer-disabled:cursor-not-allowed peer-disabled:opacity-50 data-[error=true]:text-destructive"
            data-error="false"
            data-slot="form-label"
            for="«r1»-form-item"
          >
            Description
          </label>
          <textarea
            aria-describedby="«r1»-form-item-description"
            aria-invalid="false"
            class="border-input placeholder:text-muted-foreground focus-visible:border-ring focus-visible:ring-ring/50 aria-invalid:ring-destructive/20 dark:aria-invalid:ring-destructive/40 aria-invalid:border-destructive dark:bg-input/30 flex field-sizing-content w-full rounded-md border bg-transparent px-3 py-2 text-base shadow-xs transition-[color,box-shadow] outline-none focus-visible:ring-[3px] disabled:cursor-not-allowed disabled:opacity-50 md:text-sm min-h-[100px]"
            data-slot="form-control"
            id="«r1»-form-item"
            name="description"
            placeholder="Enter task description"
          />
        </div>
        <div
          class="gap-2 flex flex-col"
          data-slot="form-item"
        >
          <label
            class="flex items-center gap-2 text-sm leading-none font-medium select-none group-data-[disabled=true]:pointer-events-none group-data-[disabled=true]:opacity-50 peer-disabled:cursor-not-allowed peer-disabled:opacity-50 data-[error=true]:text-destructive"
            data-error="false"
            data-slot="form-label"
            for="«r2»-form-item"
          >
            Deadline (Optional)
          </label>
          <div
            class="flex gap-2"
          >
            <button
              aria-controls="radix-«r3»"
              aria-describedby="«r2»-form-item-description"
              aria-expanded="false"
              aria-haspopup="dialog"
              aria-invalid="false"
              class="inline-flex items-center justify-center gap-2 whitespace-nowrap rounded-md text-sm transition-all disabled:pointer-events-none disabled:opacity-50 [&_svg]:pointer-events-none [&_svg:not([class*='size-'])]:size-4 shrink-0 [&_svg]:shrink-0 outline-none focus-visible:border-ring focus-visible:ring-ring/50 focus-visible:ring-[3px] aria-invalid:ring-destructive/20 dark:aria-invalid:ring-destructive/40 aria-invalid:border-destructive border bg-background shadow-xs hover:bg-accent hover:text-accent-foreground dark:bg-input/30 dark:border-input dark:hover:bg-input/50 h-9 px-4 py-2 has-[>svg]:px-3 w-full pl-3 text-left font-normal text-muted-foreground"
              data-slot="popover-trigger"
              data-state="closed"
              id="«r2»-form-item"
              type="button"
            >
              <span>
                Pick a date
              </span>
              <svg
                aria-hidden="true"
                class="lucide lucide-calendar ml-auto h-4 w-4 opacity-50"
                fill="none"
                height="24"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="2"
                viewBox="0 0 24 24"
                width="24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M8 2v4"
                />
                <path
                  d="M16 2v4"
                />
                <rect
                  height="18"
                  rx="2"
                  width="18"
                  x="3"
                  y="4"
                />
                <path
                  d="M3 10h18"
                />
              </svg>
            </button>
          </div>
        </div>
        <button
          class="inline-flex items-center justify-center gap-2 whitespace-nowrap rounded-md text-sm font-medium transition-all disabled:pointer-events-none disabled:opacity-50 [&_svg]:pointer-events-none [&_svg:not([class*='size-'])]:size-4 shrink-0 [&_svg]:shrink-0 outline-none focus-visible:border-ring focus-visible:ring-ring/50 focus-visible:ring-[3px] aria-invalid:ring-destructive/20 dark:aria-invalid:ring-destructive/40 aria-invalid:border-destructive bg-primary text-primary-foreground shadow-xs hover:bg-primary/90 h-9 px-4 py-2 has-[>svg]:px-3"
          data-slot="button"
          type="submit"
        >
          Create Task
        </button>
      </form>
    </div>
  </body>,
  "container": <div>
    <form
      class="space-y-4"
    >
      <div
        class="grid gap-2"
        data-slot="form-item"
      >
        <label
          class="flex items-center gap-2 text-sm leading-none font-medium select-none group-data-[disabled=true]:pointer-events-none group-data-[disabled=true]:opacity-50 peer-disabled:cursor-not-allowed peer-disabled:opacity-50 data-[error=true]:text-destructive"
          data-error="false"
          data-slot="form-label"
          for="«r0»-form-item"
        >
          Title
        </label>
        <input
          aria-describedby="«r0»-form-item-description"
          aria-invalid="false"
          class="file:text-foreground placeholder:text-muted-foreground selection:bg-primary selection:text-primary-foreground dark:bg-input/30 border-input flex h-9 w-full min-w-0 rounded-md border bg-transparent px-3 py-1 text-base shadow-xs transition-[color,box-shadow] outline-none file:inline-flex file:h-7 file:border-0 file:bg-transparent file:text-sm file:font-medium disabled:pointer-events-none disabled:cursor-not-allowed disabled:opacity-50 md:text-sm focus-visible:border-ring focus-visible:ring-ring/50 focus-visible:ring-[3px] aria-invalid:ring-destructive/20 dark:aria-invalid:ring-destructive/40 aria-invalid:border-destructive"
          data-slot="form-control"
          id="«r0»-form-item"
          name="title"
          placeholder="Task title"
          value=""
        />
      </div>
      <div
        class="grid gap-2"
        data-slot="form-item"
      >
        <label
          class="flex items-center gap-2 text-sm leading-none font-medium select-none group-data-[disabled=true]:pointer-events-none group-data-[disabled=true]:opacity-50 peer-disabled:cursor-not-allowed peer-disabled:opacity-50 data-[error=true]:text-destructive"
          data-error="false"
          data-slot="form-label"
          for="«r1»-form-item"
        >
          Description
        </label>
        <textarea
          aria-describedby="«r1»-form-item-description"
          aria-invalid="false"
          class="border-input placeholder:text-muted-foreground focus-visible:border-ring focus-visible:ring-ring/50 aria-invalid:ring-destructive/20 dark:aria-invalid:ring-destructive/40 aria-invalid:border-destructive dark:bg-input/30 flex field-sizing-content w-full rounded-md border bg-transparent px-3 py-2 text-base shadow-xs transition-[color,box-shadow] outline-none focus-visible:ring-[3px] disabled:cursor-not-allowed disabled:opacity-50 md:text-sm min-h-[100px]"
          data-slot="form-control"
          id="«r1»-form-item"
          name="description"
          placeholder="Enter task description"
        />
      </div>
      <div
        class="gap-2 flex flex-col"
        data-slot="form-item"
      >
        <label
          class="flex items-center gap-2 text-sm leading-none font-medium select-none group-data-[disabled=true]:pointer-events-none group-data-[disabled=true]:opacity-50 peer-disabled:cursor-not-allowed peer-disabled:opacity-50 data-[error=true]:text-destructive"
          data-error="false"
          data-slot="form-label"
          for="«r2»-form-item"
        >
          Deadline (Optional)
        </label>
        <div
          class="flex gap-2"
        >
          <button
            aria-controls="radix-«r3»"
            aria-describedby="«r2»-form-item-description"
            aria-expanded="false"
            aria-haspopup="dialog"
            aria-invalid="false"
            class="inline-flex items-center justify-center gap-2 whitespace-nowrap rounded-md text-sm transition-all disabled:pointer-events-none disabled:opacity-50 [&_svg]:pointer-events-none [&_svg:not([class*='size-'])]:size-4 shrink-0 [&_svg]:shrink-0 outline-none focus-visible:border-ring focus-visible:ring-ring/50 focus-visible:ring-[3px] aria-invalid:ring-destructive/20 dark:aria-invalid:ring-destructive/40 aria-invalid:border-destructive border bg-background shadow-xs hover:bg-accent hover:text-accent-foreground dark:bg-input/30 dark:border-input dark:hover:bg-input/50 h-9 px-4 py-2 has-[>svg]:px-3 w-full pl-3 text-left font-normal text-muted-foreground"
            data-slot="popover-trigger"
            data-state="closed"
            id="«r2»-form-item"
            type="button"
          >
            <span>
              Pick a date
            </span>
            <svg
              aria-hidden="true"
              class="lucide lucide-calendar ml-auto h-4 w-4 opacity-50"
              fill="none"
              height="24"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
              viewBox="0 0 24 24"
              width="24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M8 2v4"
              />
              <path
                d="M16 2v4"
              />
              <rect
                height="18"
                rx="2"
                width="18"
                x="3"
                y="4"
              />
              <path
                d="M3 10h18"
              />
            </svg>
          </button>
        </div>
      </div>
      <button
        class="inline-flex items-center justify-center gap-2 whitespace-nowrap rounded-md text-sm font-medium transition-all disabled:pointer-events-none disabled:opacity-50 [&_svg]:pointer-events-none [&_svg:not([class*='size-'])]:size-4 shrink-0 [&_svg]:shrink-0 outline-none focus-visible:border-ring focus-visible:ring-ring/50 focus-visible:ring-[3px] aria-invalid:ring-destructive/20 dark:aria-invalid:ring-destructive/40 aria-invalid:border-destructive bg-primary text-primary-foreground shadow-xs hover:bg-primary/90 h-9 px-4 py-2 has-[>svg]:px-3"
        data-slot="button"
        type="submit"
      >
        Create Task
      </button>
    </form>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
